src/OpenTelemetry/Trace/Span.cs

```diff
@@ -99,11 +99,11 @@ public sealed class Span : ISpan
 
                 if (links != null)
                 {
-                    this.links = new EvictingQueue<Link>(this.tracerConfiguration.MaxNumberOfLinks);
+                    this.links = new List<Link>(this.tracerConfiguration.MaxNumberOfLinks);
 
-                    foreach (var link in links)
+                    foreach (var link in links.Reverse().Take(this.tracerConfiguration.MaxNumberOfLinks))
                     {
-                        this.links.Add(link);
+                        this.links.Insert(0, link);
```

Comment:
same here, this is O(n) operation

Reply:
used a `for`, let's see what I'm missing next